### PR TITLE
IMath portability fixes from PostgreSQL

### PR DESCRIPTION
**Pull request by nmisch**
The PostgreSQL project carries an old copy of IMath, and I am looking to
merge in your latest version.  Some local modifications we've been
carrying belong upstream.  We arrived at these in the course of fixing
compiler warnings, not live bugs.  "Sun WorkShop 6 update 2 C 5.3"
complained about the "1 << 31" integer overflow.  I don't recall which
configuration(s) complain about passing a value of type "char" to a
ctype.h function, though I have seen it recently.

Thanks,
nm

### CODE REVIEW

**Comment by nmisch** [comment]
Can I do something to expedite this?

**Comment by nmisch** [comment]
Ping.

**Comment by nmisch** [comment]
Ping.

**Comment by nmisch** [comment]
Ping.  I've resolved merge conflicts with the 2017-06-01 change.

File: imath.c [hunk 1]
```diff
@@ -1759,7 +1759,7 @@ mp_result mp_int_read_cstring(mp_int z, mp_size radix, const char *str, char **e
     return MP_RANGE;
 
   /* Skip leading whitespace */
-  while (isspace((int)*str))
+  while (isspace((unsigned char) *str))
```
Comment (creachadair) [inline imath.c]: I think these casts to `unsigned char` should be fine, but ANSI defines these as taking `int`. Is there a specific issue you meant to address with these casts?

Comment (creachadair) [reply imath.c]: That seems a good enough reason to me. 

By the way, I'm sorry I didn't get back to this in approximately forever. Somewhere along the way I got notifications turned off on this repo, and have not been paying attention to it at all.

Comment (nmisch) [reply imath.c]: It was not urgent.  Thanks for the merge.

**Pull request merged**